Packages/com.ramtype0.meshia.mesh-simplification/Ndmf/Runtime/RendererUtility.cs

```diff
@@ -0,0 +1,40 @@
+#nullable enable
+
+using UnityEngine;
+
+namespace Meshia.MeshSimplification.Ndmf
+{
+    public class RendererUtility
+    {
+        public static Mesh? GetMesh(Renderer renderer)
+        {
+            switch (renderer)
+            {
+                case MeshRenderer meshrenderer:
+                    var meshfilter = meshrenderer.GetComponent<MeshFilter>();
+                    return meshfilter == null ? null : meshfilter.sharedMesh;
+                case SkinnedMeshRenderer skinnedMeshRenderer:
+                    return skinnedMeshRenderer.sharedMesh;
+                default:
+                    return null;
+            }
+        } 
+        
+        public static void AssignMesh(Renderer renderer, Mesh mesh)
+        {
+            switch (renderer)
+            {
+                case MeshRenderer meshrenderer:
+                    var meshfilter = meshrenderer.GetComponent<MeshFilter>();
+                    if (meshfilter == null) return;
+                    meshfilter.sharedMesh = mesh;
+                    break;
+                case SkinnedMeshRenderer skinnedMeshRenderer:
+                    skinnedMeshRenderer.sharedMesh = mesh;
+                    break;
+                default:
+                    break;
+            }
```

Comment:
```suggestion
            switch (renderer)
            {
                case MeshRenderer meshrenderer:
                    var meshfilter = meshrenderer.GetComponent<MeshFilter>();
                    if (meshfilter == null) break;
                    meshfilter.sharedMesh = mesh;
                    break;
                case SkinnedMeshRenderer skinnedMeshRenderer:
                    skinnedMeshRenderer.sharedMesh = mesh;
                    break;
                default:
                    break;
            }
            throw new ArgumentException("Could not find target property to set mesh.");
```

対象が見つからないのを単に無視するのはよくない
例外じゃないにしても警告とかは出したほうが良い

Reply:
[b2c98a7](https://github.com/RamType0/Meshia.MeshSimplification/pull/7/commits/b2c98a7ea8f96ba947e171f49bf4b796e4ce019c)

Reply:
[8e7336a](https://github.com/RamType0/Meshia.MeshSimplification/pull/7/commits/8e7336a9051862970579133535507577b3f434b0)